apps/wpcom-block-editor/src/calypso/features/gutenboarding-editor-overrides.js

```diff
@@ -54,3 +58,65 @@ function updateSettingsBar() {
 		settingsBar.prepend( saveButton );
 	} );
 }
+
+function showFirstTimeNotice() {
+	const getPreviewButton = () => document.querySelector( '.editor-post-preview' );
+
+	const awaitPreviewButton = setInterval( () => {
+		const previewButton = getPreviewButton();
+		if ( ! previewButton ) {
+			return;
+		}
+		clearInterval( awaitPreviewButton );
+
+		// Create preview link
+		// Reuse the href from existing Preview button in the header
+		// and trigger the click event handler (registered through jQuery)
+		// as seen in handlePreview() in iframe-bridge-server.js.
+		const previewLink = (
+			<a
+				href={ previewButton.href }
+				onClick={ ( e ) => {
+					e.preventDefault();
+					// Need to requery this element because the PreviewButton
+					// gets recreated when window is resized hence referencing
+					// an old destroyed button wouldn't work.
+					getPreviewButton().click();
+				} }
+				target="_top"
+			/>
+		);
+
+		// Create first time notice
+		const blockEditorHeader = document.querySelector( '.block-editor-editor-skeleton__header' );
+		const firstTimeNotice = document.createElement( 'div' );
+
+		// It seems that the close button doesn't work when the
+		// notice component is manually injected, so we'll handle
+		// the removal of this notice manually.
+		const handleRemove = () => {
+			blockEditorHeader.removeChild( firstTimeNotice );
+		};
+
+		ReactDOM.render(
```

Comment:
Would `withNotices` be canonical way of showing these instead of rendering in the header?

https://github.com/WordPress/gutenberg/blob/bfc58f53a38b9b0830198fa134a5f08eb3dad5ef/packages/components/src/higher-order/with-notices/README.md

Or did everything in core got turned into Snackbar notices? 🤔

Reply:
I tried using `withNotices`, while it did solve the close button not working, I didn't like the implementation. `withNotices` is useful if you need to handle multiple notices over time, for a single notice, it's overkill. Also, I can't render the notice immediately and needed to use `useEffect` to generate the notice once on first component mount.

Just leaving the code here for reference.

```jsx
function showFirstTimeNotice() {
	// Skip on older atomic sites.
	if ( typeof createInterpolateElement !== 'function' ) {
		return;
	}

	const getPreviewButton = () => document.querySelector( '.editor-post-preview' );

	const awaitPreviewButton = setInterval( () => {
		const previewButton = getPreviewButton();
		if ( ! previewButton ) {
			return;
		}
		clearInterval( awaitPreviewButton );

		const FirstTimeNotice = withNotices( ( { noticeOperations, noticeUI } ) => {

			// Using useEffect to create notice on component mount.
			useEffect( () => {
				// Create preview link
				// Reuse the href from existing Preview button in the header
				// and trigger the click event handler (registered through jQuery)
				// as seen in handlePreview() in iframe-bridge-server.js.
				const previewLink = (
					<a
						href={ previewButton.href }
						onClick={ ( e ) => {
							e.preventDefault();
							// Need to requery this element because the PreviewButton
							// gets recreated when window is resized hence referencing
							// an old destroyed button wouldn't work.
							getPreviewButton().click();
						} }
						target="_top"
					/>
				);

				noticeOperations.createNotice( {
					status: 'success',
					content: createInterpolateElement(
						__(
							'This is your Editor for your Home page content. To see what your site will look like when published, click <preview_link>Preview</preview_link> here or in the top bar.'
						),
						{
							preview_link: previewLink,
						}
					),
				} );
			}, [ noticeOperations ] );

			return noticeUI;
		} );

		// Create first time notice
		const firstTimeNoticeContainer = document.createElement( 'div' );

		ReactDOM.render(
			<FirstTimeNotice className="gutenboarding-editor-overrides__first-time-notice" />,
			firstTimeNoticeContainer
		);

		// Show first time notice under block header
		const blockEditorHeader = document.querySelector( '.block-editor-editor-skeleton__header' );
		blockEditorHeader.appendChild( firstTimeNoticeContainer );
	} );
}
```